code/asteroid/asteroid.cpp

```diff
@@ -1852,17 +1852,21 @@ static void asteroid_parse_section(asteroid_info *asip)
 void maybe_change_asteroid_name(asteroid_info* asip) {
 
 	SCP_string name = asip->name;
-	size_t split = name.find(' ');
+	size_t split = std::string::npos;
+
+	for (species_info species : Species_info) {
+		split = name.find(species.species_name);
+
+		if (split != std::string::npos) {
+			split = strlen(species.species_name);
+			break;
+		}
+	}
```

Comment:
This won't split correctly if the species name appears in the middle (not the beginning) of the string.

Reply:
If split is 0, then right? instead of npos

Reply:
That would work.  You could also use `compare` with a length parameter, or `strncmp` on the `c_str()`.